### ISSUE TITLE
Default upload host parameter in config/admin.php

### ISSUE THREAD

**Issue by isometriq** [z-song/laravel-admin#1010]
The default value provided in the config is `http://localhost:8000/upload/`, which may not work for everyone, if host or port number

A safer value would be `url('upload')`

Example snippet of the relevant section in the config file:
```
    'upload'  => [

        'disk' => 'admin',

        'directory'  => [
            'image'  => 'image',
            'file'   => 'file',
        ],

        'host' => url('upload'),
    ],
```

**Comment by isometriq** [comment]
This issue is more for convenience, but another reported issue is similar:
https://github.com/z-song/laravel-admin/issues/992
(about the lack of helper function usage for urls)

**Comment by isometriq** [comment]
Small detail, the disks in the `config/filesystems.php` file, can use a `url` attribute.

Maybe the disk defined in `config/admin.php` could have also the `url` attribute instead of defining `host`. Then you can get the image url with `Storage::('admin')::url('image/avatar-filename.jpg` )`

**Comment by isometriq** [comment]
I'm sorry, I just discovered I was wrong about something.

The `url()` helper method (and some other) cannot be used in the config files. The philosophy behind it being config must be stateless or static. Some discussion about this:
https://github.com/laravel/framework/issues/7671

In this discussion, a proposed way is to use the `url()` helper in a provider `register()` method. I used it like so in my `AppServiceProvider` :
```
public function register()
{
    config([
        'admin.upload.host' => url(config('admin.upload.host')),
        'filesystems.disks.public.url' =>  url(config('filesystems.disks.public.url')),
        'filesystems.disks.admin.url' =>  url(config('filesystems.disks.admin.url')),
    ]);
}
```
and left only the path portion in the config that I want to use in the `url()` helper function.
